src/contexts/CouponManagerProvider.tsx

```diff
@@ -0,0 +1,75 @@
+import {
+  createContext,
+  PropsWithChildren,
+  useCallback,
+  useContext,
+  useState,
+} from "react";
+import { CouponType } from "../types/coupon";
+
+interface CouponContextType {
+  selectedCoupon: CouponType[];
+  addCoupon: (coupon: CouponType) => void;
+  removeCoupon: (coupon: CouponType) => void;
+  isSelected: (coupon: CouponType) => boolean;
+  changeSelectedCoupon: (coupons: CouponType[]) => void;
+}
+
+const couponContext = createContext<CouponContextType>({
+  selectedCoupon: [],
+  addCoupon: () => {},
+  removeCoupon: () => {},
+  isSelected: () => false,
+  changeSelectedCoupon: () => {},
+});
+
+export function CouponManagerProvider({ children }: PropsWithChildren) {
+  const [selectedCoupon, setSelectedCoupon] = useState<CouponType[]>([]);
+
+  function addCoupon(coupon: CouponType) {
+    setSelectedCoupon((prev) => {
+      prev.push(coupon);
+      return [...prev];
+    });
```

Comment:
이런 식으로 간단하게 적어볼 수도 있을거 같네요

```suggestion
    setSelectedCoupon((prev) => {
      return [...prev, coupon];
    });
```

Reply:
맞네요!! 수정하겠습니다!

Reply:
수정 커밋: [refactor: 배열에 푸시하는 로직 간략화](https://github.com/woowacourse/react-shopping-cart/pull/396/commits/f76c62ad18335ddd01c7baaa7b6dba30e305982e)